src/sparseml/openpifpaf/trainer.py

```diff
@@ -75,6 +82,45 @@ def __init__(
             model_meta_data=model_meta_data,
         )
 
+    def _load_managers_from_checkpoint(
+        self, checkpoint_path: str, model: torch.nn.Module
+    ):
+        if checkpoint_path is None:
+            LOG.info("Not loading anything from checkpoint")
+            return
+
+        if checkpoint_path.startswith("zoo:"):
+            checkpoint_path = download_framework_model_by_recipe_type(
+                Model(checkpoint_path)
+            )
+
+        checkpoint = torch.load(checkpoint_path, map_location="cpu")
+        if "checkpoint_recipe" not in checkpoint:
+            LOG.info(f"No checkpoint recipe in checkpoint: {list(checkpoint.keys())}")
+            return
+
+        LOG.info("Found recipe in checkpoint")
+        checkpoint_manager = ScheduledModifierManager.from_yaml(
+            checkpoint["checkpoint_recipe"]
+        )
+        if checkpoint["epoch"] == -1:
+            # restore state from finished recipe
+            LOG.info(
+                "Checkpoint was from epoch -1, "
+                "checkpoint recipe is NOT overriding configured recipe"
+            )
+            checkpoint_manager.apply_structure(model, epoch=checkpoint["epoch"])
+            self.checkpoint_manager = checkpoint_manager
+        else:
+            # resume
```

Comment:
why are we choosing to override the given training recipe which was specified by the user? should at least log a warning in this case, however not convinced that this should be the default/only behavior.
What is the current "resume" rules for openpifpaf?

Reply:
That's what we did in torchvision too - was just copying that. Is that not what we want?